internal/capabilities/detector_test.go

```diff
@@ -90,66 +102,88 @@ func TestDetectCapabilities(t *testing.T) {
-			manual:   "all",
-			detected: []types.Capability{"web-scraper", "telemetry", "searchbyquery"},
-			expected: []types.Capability{"all", "web-scraper", "telemetry", "searchbyquery"},
-		},
-		{
-			name:     "Manual specific capabilities with detected",
-			manual:   "searchbyquery,getbyid",
-			detected: []types.Capability{"web-scraper", "telemetry", "searchbyprofile"},
-			expected: []types.Capability{"searchbyquery", "getbyid", "web-scraper", "telemetry", "searchbyprofile"},
+			name: "With accounts only",
+			jc: types.JobConfiguration{
+				"twitter_accounts": []string{"user:pass"},
+			},
+			expectedKeys: []string{string(teetypes.WebJob), string(teetypes.TelemetryJob), string(teetypes.TiktokJob), string(teetypes.TwitterCredentialJob), string(teetypes.TwitterJob)},
 		},
 		{
-			name:     "Overlapping manual and detected",
-			manual:   "web-scraper,custom-cap",
-			detected: []types.Capability{"web-scraper", "telemetry"},
-			expected: []types.Capability{"web-scraper", "custom-cap", "telemetry"},
+			name: "With API keys only",
+			jc: types.JobConfiguration{
+				"twitter_api_keys": []string{"key123"},
+			},
+			expectedKeys: []string{string(teetypes.WebJob), string(teetypes.TelemetryJob), string(teetypes.TiktokJob), string(teetypes.TwitterApiJob), string(teetypes.TwitterJob)},
 		},
 		{
-			name:     "Manual with spaces",
-			manual:   "cap1, cap2 , cap3",
-			detected: []types.Capability{"cap4"},
-			expected: []types.Capability{"cap1", "cap2", "cap3", "cap4"},
+			name: "With both accounts and keys",
+			jc: types.JobConfiguration{
+				"twitter_accounts": []string{"user:pass"},
+				"twitter_api_keys": []string{"key123"},
+			},
+			expectedKeys: []string{string(teetypes.WebJob), string(teetypes.TelemetryJob), string(teetypes.TiktokJob), string(teetypes.TwitterCredentialJob), string(teetypes.TwitterJob), string(teetypes.TwitterApiJob)},
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := MergeCapabilities(tt.manual, tt.detected)
+			caps := DetectCapabilities(tt.jc, nil)
+
+			jobNames := make([]string, len(caps))
+			for i, cap := range caps {
+				jobNames[i] = cap.JobType
+			}
 
-			// Sort for consistent comparison since map iteration is random
-			slices.Sort(got)
-			slices.Sort(tt.expected)
+			// Check that all expected keys are present
+			for _, expectedKey := range tt.expectedKeys {
+				found := false
+				for _, jobName := range jobNames {
+					if jobName == expectedKey {
+						found = true
+						break
+					}
+				}
+				if !found {
+					t.Errorf("Expected scraper %s not found in %v", expectedKey, jobNames)
+				}
+			}
 
-			if !reflect.DeepEqual(got, tt.expected) {
-				t.Errorf("MergeCapabilities() = %v, want %v", got, tt.expected)
+			// Check that no unexpected keys are present
+			for _, jobName := range jobNames {
+				found := false
+				for _, expectedKey := range tt.expectedKeys {
+					if jobName == expectedKey {
+						found = true
+						break
+					}
+				}
```

Comment:
Isn't this identical to the above? TBH now that I see this I think I like `DeepEqual` better (sorry for that).